docs/init.md

```diff
@@ -0,0 +1,201 @@
+   - Retrieves the correct subscription by ID from the action arguments
+   - Performs the scheduled action (payment processing, trial end, etc.)
+   - Updates subscription dates and statuses as appropriate
+
+4. The subscription status may change based on the event, which then triggers:
+   - Email notifications
+   - Payment processing
+   - Status changes and associated actions
+
+### REST API Requests
+
+For REST API requests related to subscriptions:
+
+1. WordPress identifies the request as a REST API request via `WC()->is_rest_api_request()`
+2. The core initialization flow runs
+3. WC Subscriptions checks if it's a REST API request using `wcs_is_rest_api_request()`
+4. Admin UI components are not loaded to keep the request lightweight
+5. Data handling remains consistent with standard requests, but presentation layers are skipped
+
+### AJAX Requests
+
+For AJAX requests related to subscriptions:
+
+1. WordPress identifies the request as an AJAX request via `wp_doing_ajax()`
+2. The core initialization flow runs
+3. WC Subscriptions checks if it's an AJAX request using `wcs_doing_ajax()`
+4. Depending on the specific AJAX action:
+   - Admin-specific components may be loaded for admin AJAX requests
+   - Front-end components for customer AJAX requests
+   - The response is typically JSON formatted data rather than rendered HTML
+
+## Key Differences Between Contexts
+
+| Context | Admin UI Components | Frontend Components | Template Loading | Performance Considerations |
+|---------|--------------------|--------------------|-----------------|---------------------------|
+| Admin Pages | ✅ All loaded | ❌ Not loaded | ✅ Admin templates only | Loads more components for full functionality |
+| Customer Pages | ❌ Not loaded | ✅ All loaded | ✅ Frontend templates | Focuses on cart, checkout, and my-account functionality |
+| Action Scheduler | ❌ Not loaded | ❌ Not loaded | ❌ Not needed | Lightweight, focuses on data processing only |
+| REST API | ❌ Not loaded | ⚠️ Partial loading | ❌ Not needed | Data-focused, returns structured data |
+| AJAX | ⚠️ Context-dependent | ⚠️ Context-dependent | ❌ Not needed | Minimal loading for fast responses |
+
+## Initialization Detection Functions
+
+WC Subscriptions provides helper functions to determine the current request context:
+
+```php
+// Check if current request is AJAX
+wcs_doing_ajax()
+
+// Check if current request is wp-cron (includes Action Scheduler)
+wcs_doing_cron()
+
+// Check if current request is REST API
+wcs_is_rest_api_request()
+
+// Check if current request is frontend (not admin, not AJAX, not cron, not REST)
+wcs_is_frontend_request()
+```
+
+## Performance Considerations
+
+- Admin pages load the most components and have the highest overhead
+- Action Scheduler events are optimized for performance with minimal component loading
+- REST API and AJAX requests are designed to be lightweight for responsiveness
+- Frontend pages load only what's needed for customer interactions
```

Comment:
This reads like a (slightly aspirational) statement of how things are, rather than what practices to follow in each context, or when to move work from one context to another.

Reply:
Fair enough, after exploring our init logic, I've decided to remove the section completely.